src/pybasic/pybasic.py

```diff
@@ -0,0 +1,128 @@
+"""Contains the PyBaSiC class."""
+
+import os
+from dataclasses import dataclass
+from typing import Union
+
+import numpy as np
+
+PathLike = Union[str, bytes, os.PathLike]
+ESTIMATION_MODES = ["l0"]  # NOTE convert to enum?
+
+
+@dataclass
+class Settings:
+    """A class to hold BaSiC settings.
+
+    Attributes:
+        lambda_flatfield (float)
+        estimation_mode (str)
+        max_iterations (int): Maximum number of iterations allowed in the
+            optimization.
+        optimization_tolerance (float): Tolerance of error in the optimization.
+        darkfield (bool): Whether to estimate a darkfield correction.
+        lambda_darkfield (float)
+        working_size (int)
+        max_reweight_iterations (int)
+        eplson (float)
+        varying_coeff (bool)
+        reweight_tolerance (float)
+    """
+
+    lambda_flatfield: float = 0
+    estimation_mode: str = "l0"
+    max_iterations: int = 500
+    optimization_tolerance: float = 1e-6
+    darkfield: bool = False
+    lambda_darkfield: float = 0
+    working_size: int = 0
+    max_reweight_iterations: int = 10
+    eplson: float = 0.1  # NOTE rename to epslon?
+    varying_coeff: bool = True
+    reweight_tolerance: float = 1e-3
+
+    def __post_init__(self) -> None:
+        if self.estimation_mode not in ESTIMATION_MODES:
+            raise ValueError(
+                f"Estimation mode '{self.estimation_mode}' is not valid. "
+                f"Please select mode from {ESTIMATION_MODES}."
+            )
+
+
+# NOTE how should multiple channels be handled?
```

Comment:
I think a simple implementation (a single PyBaSiC object for a single channel) might keep us from over-complication.

Reply:
I agree with @yfukai .